pyproject.toml

```diff
@@ -2,18 +2,15 @@
 requires = [
   "setuptools>=42",
   "wheel",
-  # declaring numpy versions specifically to x86_64
   # lowest NumPy we can use for a given Python,
-  # except for more exotic platform (Mac Arm flavors)
-  # Also don't allow PyPy builds
-  "numpy==1.20.0; python_version=='3.8' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
-  "numpy==1.20.0; python_version=='3.9' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
-  # As per https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
-  # safest to build at 1.21.6 for all platforms
-  "numpy==1.21.6; python_version=='3.10' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
-  "numpy==1.23.2; python_version=='3.11' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
-  # default to just numpy for unknown versions of Python
-  "numpy; python_version>='3.12'",
+  # In part adapted from: https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
+  # As per NEP29, we set the minimum version to 1.23.2 for Python <=3.11
+  # and 1.26.0 (first to support) for Python 3.12
+  "numpy==1.23.2; python_version<='3.11' and platform_python_implementation != 'PyPy'",
+  "numpy==1.26.0; python_version=='3.12' and platform_python_implementation != 'PyPy'",
+  # For unreleased versions of Python there is currently no known supported
+  # NumPy version. In that case we just let it be a bare NumPy install
+  "numpy<2.0; python_version>='3.13'",
   "scikit-build",
   "cmake",
   "cython>=0.28,<3.0",
```

Comment:
Any ideas on moving towards Cython>3.0? (I think there was chatter somewhere that the core library should move that way - at least we build with 3.0+ now).

Reply:
Should be fine, just pushed change with new cython pin.

Reply:
Looks like you're still building on Cython 0.29. Is there something else pinning things?